### ISSUE TITLE
Cholesky decomposition returns upper/lower triangular part of a matrix instead of the decomposition

### ISSUE THREAD

**Issue by CaptainSharf** [chapel-lang/chapel#16487]
### Summary of Problem
Cholesky decomposition returns the lower-triangular part of a given matrix instead of decomposing the matrix such that L'L=A. One also needs to check if the matrix A is symmetric positive definite before decomposing it into L,L'. In the current implementation the return value of `LAPACK.potrf` is ignored.

### Steps to Reproduce

**Source Code:**

```chapel
use LinearAlgebra;
var dom = {1..3,1..3};
var arr:[dom] real;
for (i,j) in dom do
    arr[i,j] = i;
for i in 1..3 do
    arr[1,i] = 0;
var matr = Matrix(arr);
var x = cholesky(matr);
writeln(x);
```

**Compile command:**
`chpl -L/usr/local/opt/lapack/lib -I/usr/local/opt/lapack/include -llapacke cholesky.chpl`

**Execution command:**
`./cholesky`

**Output**
```
Matrix is 
0.0 0.0 0.0
2.0 2.0 2.0
3.0 3.0 3.0
L is 
0.0 0.0 0.0
2.0 2.0 0.0
3.0 3.0 3.0
```
The cholesky decomposition of the matrix does not exist as it is singular
### Configuration Information
- `chpl --version: chpl version 1.23.0`

**Comment by ben-albrecht** [comment]
@CaptainSharf - Thanks for reporting!

I agree that checking the return value of `potrf` and halting with a helpful error message if not `0` should resolve this issue.

**Comment by CaptainSharf** [comment]
This issue has been fixed as part of the PR #16514
